cmd/tsconnect/wasm/wasm_js.go

```diff
@@ -324,6 +350,117 @@ func (i *jsIPN) logout() {
 	go i.lb.Logout()
 }
 
+func (i *jsIPN) tcp(host string, port int, readCallback js.Value, connectTimeoutSeconds int) any {
+	portInt := strconv.Itoa(port)
+	addr := net.JoinHostPort(host, portInt)
+
+	return makePromise(func() (any, error) {
+		if readCallback.Type() != js.TypeFunction {
+			return nil, errors.New("Invalid readCallback received")
+		}
+
+		connectTimeout := time.Duration(connectTimeoutSeconds) * time.Second
+		jsTCPSession, err := i.NewJSTCPSession(addr, readCallback, connectTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		jsObjectWrapper := jsTCPSession.jsWrapper()
+		return jsObjectWrapper, nil
+	})
+
+}
+
+type jsTCPSession struct {
+	jsIPN       *jsIPN
+	conn        net.Conn
+	writeBuffer []byte
+	readBuffer  []byte
+}
+
+const defaultTCPConnectTimeout time.Duration = 5 * time.Second
+
+func (i *jsIPN) NewJSTCPSession(addr string, readCallback js.Value, connectTimeout time.Duration) (*jsTCPSession, error) {
+	if connectTimeout == 0 {
+		connectTimeout = defaultTCPConnectTimeout
+	}
```

Comment:
I'd prefer the caller do this, at the first point it goes from optional float64 seconds to a time.Duration.

So we never have a magic meaning of a time.Duration value (0).